parcel/parcel.jl

```diff
@@ -67,23 +69,48 @@ function parcel_model(dY, Y, p, t)
 
     # TODO - we should zero out all tendencies and augemnt them
     # TODO - add immersion, homogeneous, ...
-    #dN_act_dt_immersion = FT(0)
     #dN_act_dt_homogeneous = FT(0)
 
     dN_act_dt_depo = FT(0)
+    dqi_dt_new_depo = FT(0)
     if "DustDeposition" in ice_nucleation_modes
         AF = CMI_het.dust_activated_number_fraction(
             prs,
             S_i,
             T,
-            CMT.DesertDustType(),
+            aerosol_type,
         )
         dN_act_dt_depo = max(FT(0), AF * N_aer - N_ice) / const_dt
+        dqi_dt_new_depo = dN_act_dt_depo * 4 / 3 * π * r_nuc^3 * ρ_ice / ρ_air
+    end
+
+    dN_act_dt_immersion = FT(0)
+    dqi_dt_new_immers = FT(0)
+    if "ImmersionFreezing" in ice_nucleation_modes
+        Δa_w = T > FT(185) && T < FT(235) ?
+            CMO.a_w_xT(H2SO4_prs, thermo_params, x_sulph, T) - CMO.a_w_ice(thermo_params, T) :
+            CMO.a_w_eT(thermo_params, e, T) - CMO.a_w_ice(thermo_params, T)
+        J_immersion = CMI_het.ABIFM_J(prs, aerosol_type, Δa_w)
+        if "Monodisperse" in droplet_size_distribution && "ImmersionFreezing" in ice_nucleation_modes
+            r_l = cbrt(q_liq / N_liq / (4 / 3 * π) / ρ_liq * ρ_air)
+            A_aer = 4 * π * r_l^2 * N_liq
```

Comment:
Why do we have to multiply by N_liq here?

Reply:
To get the total surface area to multiply J to. If it makes more sense, I can move it to line 97 so it would be `dN_act_dt_immersion = max(FT(0), J_immersion * A_aer * N_liq)`?

Reply:
So maybe we have to update the documentation. Because I think there we say it's the individual particle area? 

https://clima.github.io/CloudMicrophysics.jl/dev/IceNucleation/#ABIFM-for-Sulphuric-Acid-Containing-Droplets

Reply:
Right, lets move it to 97. This way it will be more similar to the docs